### PR TITLE
Initial implementation of torrent.py

### PR DESCRIPTION
**Pull request by badstreff**

### CODE REVIEW

**Comment by badstreff** [comment]
Please discuss, everything that we discussed should be there for taking a torrent_dict returned by the bencode decode function and creating a torrent.

I used https://wiki.theory.org/BitTorrentSpecification#Metainfo_File_Structure for determining which fields to add as optional and which are required.

**Comment by virtualhealthcitizen** [comment]
Hello BadStreff. 👋 

I've reviewed your changes. 

Wow!

Amazing. 

I think this looks good!  👀

I like the fact that we have a good idea now about which torrent fields are required and which are optional. It puts development tasks into perspective for this particular .py file.

Thanks for doing this! 🌮

**Comment by virtualhealthcitizen** [comment]
Also, I'll keep a tab open for https://wiki.theory.org/BitTorrentSpecification#Metainfo_File_Structure. 
Very useful for this part.

**Comment by badstreff** [comment]
@squidmin No problem, also thanks for catching that issue with the created_by assignment. I resolved it in the latest commit.

File: debug/torrent.py [hunk 1]
```diff
@@ -32,8 +32,8 @@ def __init__(self, torrent_dict):
```
Comment (virtualhealthcitizen) [inline debug/torrent.py:32]: Looks good!

**Pull request merged**